### ISSUE TITLE
ColumnString avoid memory copy

### ISSUE THREAD

**Issue by 1261385937** [ClickHouse/clickhouse-cpp#197]
```c++
void ColumnString::Append(std::string_view str) {
    if (blocks_.size() == 0 || blocks_.back().GetAvailble() < str.length())
    {
        blocks_.emplace_back(std::max(DEFAULT_BLOCK_SIZE, str.size()));
    }

    items_.emplace_back(blocks_.back().AppendUnsafe(str));
}
```
Always cpoy the str into the Block, this is not necessary. We can move the data into `std::queue<std::string>` if the str is not useful any more.
I remeber a previous version use the  `std::vector<std::string>` to hold the str, but also use the cpoy. The code like :
```c++
void ColumnString::Append(const std::string& str) {
       data_.push_back(str)
}
```
some advantage that i missed ?

**Comment by Enmk** [comment]
IMO we can add `ColumnString::Append(std::string && str)`.

That would steal ownership of data and put it into some internal container, like `std::list<std::string>`, and add reference to `ColumnString::items_`.
We don't need `std::deque` here, since `operator[]` should use `ColumnString::items_` for random access;

That way we can have both fast de-serealization (thanks to amortized memory allocations with `ColumnString::Block`) and more efficient `Append`ing.

Alternatively, if string data lifetime is managed elsewhere and guaranteed to outlive the `Block` sent to the server, we can introduce `ColumnString::AppendNoCopy(std::string_view)`, that would just update `ColumnString::items_`.

**Comment by Enmk** [comment]
Closed in #202
